baseplate/__init__.py

```diff
@@ -629,7 +629,10 @@ def finish(self, exc_info: Optional[_ExcInfo] = None) -> None:
 
         """
         for observer in self.observers:
-            observer.on_finish(exc_info)
+            try:
+                observer.on_finish(exc_info)
+            except Exception:
+                logger.exception("Exception raised while finalizing observer")
```

Comment:
Please make it so this exception includes all the details the original exception would include. I don't want things to become harder to debug because we're catching this exception.

Reply:
`logger.exception` actually adds that information for us
https://docs.python.org/3/library/logging.html#logging.Logger.exception